test/jdk/tools/jlink/basic/AllModulePath.java

```diff
@@ -137,7 +164,120 @@ public void testAddModules() throws Throwable {
+    }
+
+    /*
+     * --module-path not-exist and --add-modules ALL-MODULE-PATH is an error.
+     */
+    @Test
+    public void modulePathEmpty() {
+        if (isExplodedJDKImage()) {
+            return;
+        }
+        Path targetPath = HELPER.createNewImageDir("all-mod-path-not-existing");
+        String strNotExists = "not-exist";
+        Path notExists = Path.of(strNotExists);
+        if (Files.exists(notExists)) {
+            throw new RuntimeException("Test setup error, path must not exist!");
+        }
+        List<String> allArgs = List.of("--add-modules", "ALL-MODULE-PATH",
+                                       "--module-path", notExists.toString(),
+                                       "--output", targetPath.toString());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintWriter out = new PrintWriter(baos);
+        ByteArrayOutputStream berrOs = new ByteArrayOutputStream();
+        PrintWriter err = new PrintWriter(berrOs);
+        int rc = JLINK_TOOL.run(out, err, allArgs.toArray(new String[] {}));
+        assertTrue(rc != 0);
+        String actual = new String(baos.toByteArray()).trim();
+        assertTrue(actual.startsWith("Error: No module found in module path"));
+        assertTrue(actual.contains(strNotExists));
+    }
+
+    /*
+     * --add-modules ALL-MODULE-PATH with an existing module path and module
+     * limits applied.
+     */
+    @Test
+    public void modulePathWithLimitMods() throws Exception {
+        if (isExplodedJDKImage()) {
+            return;
+        }
+        Path targetPath = HELPER.createNewImageDir("all-mods-limit-mods");
+        String moduleName = "com.baz.runtime";
+        Result result = HELPER.generateDefaultJModule(moduleName, "jdk.jfr");
+        Path customModulePath = result.getFile().getParent();
+        List<String> allArgs = List.of("--add-modules", "ALL-MODULE-PATH",
+                                       "--limit-modules", "jdk.jfr", // A dependency of com.baz.runtime
+                                       "--module-path", customModulePath.toString(),
+                                       "--output", targetPath.toString());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintWriter out = new PrintWriter(baos);
+        ByteArrayOutputStream berrOs = new ByteArrayOutputStream();
+        PrintWriter err = new PrintWriter(berrOs);
+        int rc = JLINK_TOOL.run(out, err, allArgs.toArray(new String[] {}));
+        assertTrue(rc == 0);
+        String stdOut = new String(baos.toByteArray());
+        String stdErr = new String(berrOs.toByteArray());
+        assertTrue(stdOut.isEmpty());
+        assertTrue(stdErr.isEmpty());
+        List<String> expected = List.of("java.base", "jdk.jfr");
+        verifyListModules(targetPath, expected);
+    }
+
+    /*
+     * --add-modules ALL-MODULE-PATH with an existing module-path.
+     */
+    @Test
+    public void modulePath() throws Exception {
```

Comment:
Any difference to `testAllModulePath` which does:

```
jlink --output image --add-modules ALL-MODULE-PATH --module-path mods
```

Reply:
No difference. I've dropped it.